agent/agent.go

```diff
@@ -207,7 +207,7 @@ func (a *Agent) run(ctx context.Context) {
 
 			// select a session registration delay from backoff range.
 			delay := time.Duration(rand.Int63n(int64(backoff)))
-			session = newSession(ctx, a, delay)
+			session = newSession(ctx, a, delay, session.sessionID)
```

Comment:
Any cases you can think of where this might not be a good idea? I can't think of it right now.